### ISSUE TITLE
Android 15 + SDK 35 edge-to-edge support?

### ISSUE THREAD

**Issue by blakef** [AppAndFlow/react-native-safe-area-context#552]
Is this supported by the library?  We were playing around with it, and it doesn't seem to work out of the box:

## Community Template:
![CleanShot 2024-11-13 at 10 31 00@2x](https://github.com/user-attachments/assets/39fabf3f-4641-42e3-949d-a33170225d00)

## Community Template + RNSAC:
![CleanShot 2024-11-12 at 21 13 15@2x](https://github.com/user-attachments/assets/608e2e97-07ee-4d62-9d5d-e4c1dc56d942)

# Code:
```bash
npx @react-native-community/cli init SafeAreaViewExample
```
Then I just swapped out the `SafeAreaView` with rnsac:
```diff
diff --git a/App.tsx b/App.tsx
index 125fe1b..e71a529 100644
--- a/App.tsx
+++ b/App.tsx
@@ -8,7 +8,6 @@
 import React from 'react';
 import type {PropsWithChildren} from 'react';
 import {
-  SafeAreaView,
   ScrollView,
   StatusBar,
   StyleSheet,
@@ -24,6 +23,7 @@ import {
   LearnMoreLinks,
   ReloadInstructions,
 } from 'react-native/Libraries/NewAppScreen';
+import {SafeAreaProvider, SafeAreaView} from 'react-native-safe-area-context';
 
 type SectionProps = PropsWithChildren<{
   title: string;
@@ -63,6 +63,7 @@ function App(): React.JSX.Element {
   };
 
   return (
+  <SafeAreaProvider>
     <SafeAreaView style={backgroundStyle}>
       <StatusBar
         barStyle={isDarkMode ? 'light-content' : 'dark-content'}
@@ -93,6 +94,7 @@ function App(): React.JSX.Element {
         </View>
       </ScrollView>
     </SafeAreaView>
+  </SafeAreaProvider>
   );
 }
 
diff --git a/android/build.gradle b/android/build.gradle
index a9ea023..2320d03 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -3,7 +3,7 @@ buildscript {
         buildToolsVersion = "35.0.0"
         minSdkVersion = 24
         compileSdkVersion = 35
-        targetSdkVersion = 34
+        targetSdkVersion = 35
         ndkVersion = "26.1.10909125"
         kotlinVersion = "1.9.24"
     }
diff --git a/package.json b/package.json
index ad9a85e..a7ca2b7 100644
--- a/package.json
+++ b/package.json
@@ -11,7 +11,8 @@
   },
   "dependencies": {
     "react": "18.3.1",
-    "react-native": "0.76.1"
+    "react-native": "0.76.1",
+    "react-native-safe-area-context": "^4.14.0"
   },
   "devDependencies": {
     "@babel/core": "^7.25.2",
@@ -36,4 +37,4 @@
   "engines": {
     "node": ">=18"
   }
-}
\ No newline at end of file
+}

```

**Comment by jmaxt12** [comment]
I'm having similar issues. Using the same device, I'm getting a lot cut off. On android 14, things look and work great

**Comment by patrikduksin** [comment]
I'm having same issue!

**Comment by amkiran** [comment]
Issue occurs for me as well. any workaround until we get a fix?

**Comment by joshpensky** [comment]
We're also running into this issue! 

Our hack for now is to just double the top inset on SDK 35, but a more robust solution would of course be appreciated 🙂

**Comment by LeunensMichiel** [comment]
I've updated my Android SDK (new 35 revision) and now it seems to magically work. Don't know for sure though if its related to that.

**Comment by aarontuftss** [comment]
+ 1

**Comment by jacquesjg** [comment]
Works for me on older version 4.3.4 when using an Activity with windowDrawsSystemBarBackground = true.

Worked with both navigation styles, too.

However, on an activity with different settings, I did notice very strange behavior with insets having values, and then resetting to zero.

**Comment by jacquesjg** [comment]
I always have insets when wrapping the app in a React Native `Modal` and set `statusBarsTranslucent` to true. :)

**Comment by gavrichards** [comment]
[This related issue](https://github.com/AppAndFlow/react-native-safe-area-context/issues/546#issuecomment-2641271816) was just closed as a duplicate of this. Has anyone made any progress with a workaround? I'm surprised how little noise there is in the community about this issue when it must surely be having a widespread impact on apps.

**Comment by janicduplessis** [comment]
Is this still happening? I just tested with a fresh template with version 0.78.0 of react native and 5.2.0 of react-native-safe-area-context and it seems to work fine. Maybe as mentioned previously also make sure the android sdk 35 is up to date.

**Comment by dprevost-LMI** [comment]
🤔 After a bit of digging, in one case, I was excluding the bottom edge I wanted the safe area since before RN 77 + Android 15 something else was adding bottom padding too. So now it works properly.

Another case I need to figure out is with the `@expo/react-native-action-sheet` opening at the bottom.

Conclusion: It would make sense that it is fixed now!

**Comment by janicduplessis** [comment]
Let’s close this and we can reopen if someone can repro reliably.

**Comment by Yurii-Lutsyk** [comment]
No updates, they just don't care

**Comment by dprevost-LMI** [comment]
I just want to point out that for me and @janicduplessis, that seems to be working fine. If there are still cases not fixed, with some detailed video and screenshot, maybe it would be worth further investigation. And since this bug is closed, gathering all those details in a new issue with a reproducible example might have more traction!

**Comment by Yurii-Lutsyk** [comment]
> I just want to point out that for me and [@janicduplessis](https://github.com/janicduplessis), that seems to be working fine. If there are still cases not fixed, with some detailed video and screenshot, maybe it would be worth further investigation. And since this bug is closed, gathering all those details in a new issue with a reproducible example might have more traction!

Thank you!

**Comment by ngocdinhalobridge** [comment]
any update? :(

**Comment by patrikduksin** [comment]
@ngocdinhalobridge it's fixed already, just use latest version

**Comment by iway1** [comment]
Just to make it clear for others:

- Upgrade to newer version of react-native (Seems like > 0.78 works)
- Upgrade to newest compatible version of `react-native-safe-area-context`
- Target android SDK 35+

Doesn't seem like anyone has tested this super thoroughly to figure out which of above conditions are exactly necessary but if you do the above 3 you should be good, `insets.bottom` will begin to have a non zero value due to edge to edge causing UI to go under the bottom bar.

**Comment by amkiran** [comment]
> Just to make it clear for others:
> 
> * Upgrade to newer version of react-native (Seems like >= 0.78 works)
> * Upgrade to newest compatible version of `react-native-safe-area-context`
> * Target android SDK 35+
> 
> Doesn't seem like anyone has tested this super thoroughly to figure out which of above conditions are exactly necessary but if you do the above 3 you should be good, `insets.bottom` will begin to have a non zero value due to edge to edge causing UI to go under the bottom bar.

Yes, This should work. I did this last month and worked on most devices.
But I believe there were still issues on older Android devices (don't recall which ones)
